src/main/java/Quill.java

```diff
@@ -0,0 +1,127 @@
+import quill.exception.EmptyDateException;
+import quill.exception.QuillException;
+import quill.task.Deadline;
+import quill.task.Event;
+import quill.task.Task;
+import quill.task.Todo;
+
+import java.util.Scanner;
+public class Quill {
+    public static final String horizontalLine = "____________________________________________________________";
+    public static void printAddTask(Task tasks) {
+        System.out.println(horizontalLine + "\nGot it. I've added this task:");
+        System.out.println(tasks.toString());
+        System.out.println("Now you have " + Task.getTotalTasks() + " tasks in the list.\n" + horizontalLine);
+    }
+
+    public static void performMarkOrUnmark(String line, Task[] tasks, boolean isDone) {
+        try {
+            int taskNumber = Integer.parseInt(line) - 1;
+            if (isDone) {
+                tasks[taskNumber].markAsDone();
+                System.out.println(horizontalLine + "\nNice! I've marked this task as done:");
+            } else {
+                tasks[taskNumber].markAsNotDone();
+                System.out.println(horizontalLine + "\nOK, I've marked this task as not done yet:");
+            }
+            System.out.println(tasks[taskNumber].toString() + "\n" + horizontalLine);
+        } catch (NullPointerException e) {
+            System.out.println("Hey, wake up! That task? Non-existent. Try something real.");
+        } catch (ArrayIndexOutOfBoundsException e) {
+            System.out.println("1 to 100 only. Seriously, no zeros, no hundreds. Got it?");
+        } catch (NumberFormatException e) {
+            System.out.println("Listen up! Numbers only, got it? Don't bother with anything else");
+        }
+    }
+
+    public static void main(String[] args) throws QuillException {
```

Comment:
You could consider refactoring this method to separate methods to adhere to this code quality [guideline](https://nus-cs2113-ay2324s2.github.io/website/se-book-adapted/chapters/codeQuality.html#avoid-long-methods)